FullStackMVCAPP/FullStackMVCAPP/Controllers/CastleController.cs

```diff
@@ -12,11 +15,12 @@ public class CastleController : Controller
         // GET: Castle
         public ActionResult Index()
         {
-            using (var db = new GOTContext()) 
-            {
-                var castleList = db.Castles.ToList();
-                return View(castleList);
-            }
+            GOTContext  context = new DataContext.GOTContext();
+            CastleRepository castleRepository = new CastleRepository();
+
+            CastleService castleService = new CastleService(context, castleRepository);
```

Comment:
This is creating the service class in the controller however, it would be better to use an interface here. I can't see any interfaces for the services only for the repositories. Using the interface would allow you to shorten these two lines of code into something like this:
```
var listOfAllCastles = _castleService.GetMeEveryCastle();
return View(listOfAllCastles);
```